framework/src/main/java/org/checkerframework/common/returnsreceiver/qual/This.java

```diff
@@ -0,0 +1,39 @@
+package org.checkerframework.common.returnsreceiver.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.PolymorphicQualifier;
+
+/**
+ * Write {@code @This} on the return type of a method that always returns its receiver ({@code
+ * this}). For example:
+ *
+ * <pre><code>
+ * class MyBuilder {
+ *   &#064;This MyBuilder setName(String name) {
+ *     this.name = name;
+ *     return this;
+ *   }
+ * }
+ * </code></pre>
+ *
+ * Strictly speaking, this is a polymorphic annotation, but when you write it on a return value, the
+ * Returns Receiver Checker automatically adds it to the receiver, so the above method is equivalent
```

Comment:
Instead of putting it so operationally, why not make this simply a default for all receivers.
Is there any harm if they are always `@This`?
Is there any harm in allowing programmers to explicitly mark receivers `@This`?

Reply:
In terms of explicitly marking receivers `@This`, I think that just adds confusion with not much gain (probably most programmers don't even know you can annotate the receiver).  But let's see, maybe we'll be able to make it default if we can work out the implementation issues.

Reply:
I would see it the other way around: why forbid something that other type systems allow? Conceptually, you want that both the receiver and return type are annotated `@This`. So if somebody wants to make that explicit, what's the harm? Most users won't care about this, I agree, but it would make the type system follow the common style of the framework.

Reply:
@wmdietl what if a programmer writes `@This` on the receiver but not the return value?  Should we add code to insert `@This` on the return value in that case?

Reply:
I would add `@This` on the return value exactly when it is currently, regardless of whether `@This` is explicitly or implicitly written on the receiver.

Reply:
In my mind, the receiver parameter is always `@This`. The return type is only `@This` if it is a special method that actually returns the receiver.
So no, I don't think there should be logic that adds `@This` to the return type if the receiver has it.